### ISSUE TITLE
Question : Is there a method that allows an icon to be flip/flopped when a folder is opened/closed?

### ISSUE THREAD

**Issue by davidkjackson54** [microsoft/vscode#128326]
I have an VSCode Explorer Extension that makes us of a TreeView. 
The Treeview basically has the equivalent of folders and files .  I have icons for folder, folder-open and files.  
When the folder is closed I set the approprate icon. When the fodler is expanded, I want to change the icon to the folder-open icon  etc. 
I trap the open and close of the folder by using the onDidExpandElement and onDidCollapseElement events - where I change the iconPath to the relevant icon.
 
 My problem is then redisplaying that element in the tree in order to flip the icon.  If I perform a node level refresh by invoking onDidChangeTreeData, it causes all of that nodes children  to be rebuilt  even if the folder is being closed  which is expensive and takes time.
 
 I know that File Icon Themes  appear to be able to very easily flip flop an icon so there must be a way of accomplishing this but I can find no supporting doc on how to accomplish what would seem to be a relative simple task? 
 
 Basically, it comes down to how do you change an icon in the Treeview without having to trigger an onDidChangeTreeData event which really complicates the issue. Data is not changing - only the icon.

**Comment by isidorn** [comment]
Forwarding to @alexr00 since she owns the custom tree experience

**Comment by alexr00** [comment]
The current API doesn't provide a way to do this.

**Comment by davidkjackson54** [comment]
Why has this been closed before we have had chance to respond? 
If it isn't availbale hten make it an Enhancement request.
